src/freon.c

```diff
@@ -0,0 +1,186 @@
+/* Copyright (c) 2015 The crouton Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ *
+ * LD_PRELOAD hack to make Xorg happy in a system without VT-switching.
+ * gcc -shared -fPIC  -ldl -Wall -O2 freon.c -o croutonfreon.so
+ *
+ * Powered by black magic.
+ */
+
+#define _GNU_SOURCE
+#include <dlfcn.h>
+#include <stdio.h>
+#include <sys/file.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <string.h>
+#include <stdarg.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <linux/input.h>
+#include <linux/vt.h>
+
+#define LOCK_FILE_DIR "/tmp/crouton-lock"
+#define DISPLAY_LOCK_FILE LOCK_FILE_DIR "/display"
+#define FREON_DBUS_METHOD_CALL(function) \
+    system("host-dbus dbus-send --system --dest=org.chromium.LibCrosService " \
+           "--type=method_call --print-reply /org/chromium/LibCrosService " \
+           "org.chromium.LibCrosServiceInterface." #function)
+
+// #define TRACE(...) fprintf(stderr, __VA_ARGS__)
+#define TRACE(...)
+#define ERROR(...) fprintf(stderr, __VA_ARGS__)
+
+static int tty0fd = -1;
+static int tty7fd = -1;
+static int lockfd = -1;
+
+static int (*orig_ioctl)(int d, int request, void* data);
+static int (*orig_open)(const char *pathname, int flags, mode_t mode);
+static int (*orig_close)(int fd);
+
+static void preload_init() {
+    orig_ioctl = dlsym(RTLD_NEXT, "ioctl");
+    orig_open = dlsym(RTLD_NEXT, "open");
+    orig_close = dlsym(RTLD_NEXT, "close");
+}
+
+static int set_display_lock(unsigned int pid) {
```

Comment:
Add a short description of the function, stating that pid == 0 is a special case.